src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php

```diff
@@ -100,30 +169,37 @@ private function isInlineableDefinition($id, Definition $definition, ServiceRefe
             return false;
         }
 
-        if (!$graph->hasNode($id)) {
+        if (!$this->graph->hasNode($id)) {
             return true;
         }
 
         if ($this->currentId == $id) {
             return false;
         }
+        $this->connectedIds[$id] = true;
 
-        $ids = array();
-        foreach ($graph->getNode($id)->getInEdges() as $edge) {
+        $srcIds = array();
+        $srcIdsCount = 0;
+        foreach ($this->graph->getNode($id)->getInEdges() as $edge) {
+            $srcId = $edge->getSourceNode()->getId();
+            $this->connectedIds[$srcId] = true;
             if ($edge->isWeak()) {
                 return false;
             }
-            $ids[] = $edge->getSourceNode()->getId();
+            $srcIds[$srcId] = true;
+            ++$srcIdsCount;
         }
 
-        if (count(array_unique($ids)) > 1) {
+        if (1 !== \count($srcIds)) {
```

Comment:
what is the difference between `\count($srcIds)` and `$srcIdsCount` ? If `$srcIdsCount` is the number of references (and so a single service referencing it multiple times is counted multiple times), the variable name is not clear.